nixos/modules/services/databases/postgresql.nix

```diff
@@ -472,12 +485,16 @@ in
       '';
     }) cfg.ensureUsers;
 
+    warnings = lib.optional (cfg.enableTCPIP == true) ''
+      Behaviour of `services.postgresql.enableTCPIP` changed from binding on all interfaces to binding on localhost in addition to the unix socket.
```

Comment:
This is wrong: there's now an option called `listenAddresses` that controls where to listen.

Generally, I'm wondering if we shouldn't just deprecate the `enableTCPIP` option and give users fine-grained control over the listen address part.

In fact, we don't even need a `listenAddresses` option, we can just define the option in the freeform settings type (I mean `settings.listen_addresses`).

I.e.

* `options.services.postgresql.enableTCPIP` gives a deprecation warning.
* if `enableTCPIP` is set, keep the old behavior (and remove it after a while - given that this is a rather central piece, we may want to go that route).
* when it's `null`, don't touch `listen_addresses`.

Does that make sense?